lib/sneakers/handlers/maxretry.rb

```diff
@@ -59,23 +66,25 @@ def initialize(channel, queue, opts)
         Sneakers.logger.debug do
           "#{log_prefix} creating queue=#{retry_name} x-dead-letter-exchange=#{requeue_name}"
         end
-        @retry_queue = @channel.queue(retry_name,
-                                     :durable => queue_durable?,
-                                     :arguments => {
+
+        @retry_queue = @channel.queue(retry_queue_name,
+                                      :durable => queue_durable?,
+                                      :arguments => {
                                        :'x-dead-letter-exchange' => requeue_name,
-                                       :'x-message-ttl' => @opts[:retry_timeout] || 60000
-                                     })
-        @retry_queue.bind(@retry_exchange, :routing_key => '#')
+                                       :'x-message-ttl' => @opts[:retry_timeout] || 60000,
+                                       :'x-dead-letter-routing-key' => requeue_routing_key
```

Comment:
This will change the current behavior I believe. My understanding of this setting is that if it is unset, the routing-key from the originally published message will be used in the retry queue. When the TTL is hit and it goes to the dead-letter exchange (which will then forward it onto the original queue sneakers is subscribing to), it will again pass along the original message's routing key. By setting the value, you're over-writing the original message's routing key, potentially causing the retried message to be different than when it first went through the queue.

I'm not an expert in this area of Rabbit, as we never don't use routing keys, but I suspect this will cause some configurations to not work. I think a test case of this would be to setup your Sneakers worker to use a queue that binds to a topic exchange with a rule to only get messages that have a routing key of "hello". Then publish a message with that routing key, have your worker fail it so it goes through this retry logic, and set the value for this requeue routing key to be "world" and see if you get it again in your worker. If you do, I think a further validation would be to verify that that message you're getting has the routing key value of "hello" and not "world". My suspicion is that you'll never see the retry and if you do, it will have a routing key value of "world".